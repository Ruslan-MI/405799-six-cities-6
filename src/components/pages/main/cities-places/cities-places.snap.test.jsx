import React from "react";
import {
  render,
} from "@testing-library/react";
import CitiesPlaces from "./cities-places";
import {
  StoreNameSpace,
} from "../../../../const";
import Test, {
  mockDataInitialState,
  mockMainPageInitialState,
  mockOffers,
} from "../../../../mocks-for-test";

it(`Cities places should render correctly`, () => {
  const dataForStore = {
    [StoreNameSpace.MAIN_PAGE]: {
      ...mockMainPageInitialState,
      currentCity: mockOffers[0].city.name,
    },
    [StoreNameSpace.DATA]: {
      ...mockDataInitialState,
      offers: mockOffers,
    },
  };

  const {
    container,
  } = render(<Test dataForStore={dataForStore}>
    <CitiesPlaces />
  </Test>);

  expect(container).toMatchSnapshot();
});
