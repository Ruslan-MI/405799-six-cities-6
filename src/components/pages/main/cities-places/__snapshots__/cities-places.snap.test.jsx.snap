// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities places should render correctly 1`] = `
<div>
  <section
    class="cities__places places"
  >
    <h2
      class="visually-hidden"
    >
      Places
    </h2>
    <b
      class="places__found"
    >
      1
       
      place
       to stay in 
      Amsterdam
    </b>
    <form
      action="#"
      class="places__sorting"
      method="get"
    >
      <span
        class="places__sorting-caption"
      >
        Sort by 
      </span>
      <span
        class="places__sorting-type"
        tabindex="0"
      >
        Popular
        <svg
          class="places__sorting-arrow"
          height="4"
          width="7"
        >
          <use
            xlink:href="#icon-arrow-select"
          />
        </svg>
      </span>
      <ul
        class="places__options places__options--custom "
      >
        <li
          class="places__option places__option--active"
          tabindex="0"
        >
          Popular
        </li>
        <li
          class="places__option "
          tabindex="0"
        >
          Price: low to high
        </li>
        <li
          class="places__option "
          tabindex="0"
        >
          Price: high to low
        </li>
        <li
          class="places__option "
          tabindex="0"
        >
          Top rated first
        </li>
      </ul>
    </form>
    <div
      class="cities__places-list places__list tabs__content"
    >
      <article
        class="cities__place-card
                        
                         place-card"
        data-testid="place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper
                      
                       place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="img/1.png"
              width="260"
            />
          </a>
        </div>
        <div
          class=" place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                120
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button "
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                $
                To
                 bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 100%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              Beautiful & luxurious studio at great location
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            apartment
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
